core/includes/config.inc

```diff
@@ -296,10 +296,13 @@ function config_install_default_config($module, $config_name = NULL) {
         // We only create new configs, and do not overwrite existing ones.
         if ($config->isNew()) {
           $config->setData($data);
+          module_invoke_all('config_create', $config);
           $config->save();
         }
       }
     }
+
+    backdrop_flush_all_caches();
```

Comment:
The commit message for adding this line says "Fixing missing admin bar".

Is that a problem somehow created by the invoking of `hook_config_create()`, or was it an existing issue when enabling new modules?

`backdrop_flush_all_caches()` is pretty intensive and would happen for every individual module enabled (if multiple were enabled at once). `backdrop_flush_all_caches()` is already called after enabling and disabling the entire set of modules in `system_modules_submit()`.

Could this be made more targeted or elaborate on why it's needed?

Reply:
I removed this in testing and could not find a consequence of it. I pushed up a commit for now without it for further testing.

Reply:
We were having a weird thing where admin bar disappeared (consistently) after syncing a field and instance. Tried lots of things, thought that that cache clear fixed it, but forgot to ask @stpaultim to test that. Wasnt meant to be permanent, I suspect what we needed was an admin bar cache clear, but that error was a bit frustrating.

Reply:
Okay, I hope it's fine that I removed it then. We have lots of places where the admin bar cache needs to be flushed, including language updates, menu links, vocabularies, content types... maybe we should add `admin_bar_config_create()` to flush the caches on new config creation, or `admin_bar_enable_modules()` to flush on module enable (though I think all caches are already flushed after module enable).

Reply:
I tested with Tim's test recipe on my local site, and the admin bar appeared for me after installing the module on my local site.